enhancements/console/customize-perspectives.md

```diff
@@ -0,0 +1,279 @@
+---
+title: customize-perspectives
+authors:
+  - "@lokanandaprabhu"
+reviewers:
+  - "@jerolimov"
+  - "@spadgett"
+approvers:
+  - "@spadgett"
+  - "@cvogt"
+api-approvers:
+  - "@jerolimov"
+  - "@cvogt"
+creation-date: 2022-07-26
+last-updated: 2022-08-18
+tracking-link: 
+  - https://issues.redhat.com/browse/ODC-6695
+status: implementable
+---
+
+# Customize Perspectives 
+
+## Summary
+
+This document provides details about how cluster administrator 
+can hide user perspectives. 
+
+## Motivation
+
+Cluster administrator wants to hide the "Administrator" perspective 
+from non-privileged or all users and to hide the "Developer" 
+perspective for all users.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I want to hide "Administrator" perspective 
+from all non-privileged users. Non-privileged users are users that cannot get
+all `namespaces` (based on RBAC check)
+
+#### Story 2
+
+As a cluster admin, I want to hide "Developer" perspective
+for all users.
+
+#### Story 3
+
+As a cluster admin, I want to hide any perspective contributed by a plugin.
+
+### Goals
+
+- Allow cluster admins to hide the user perspectives.
+- Provide a way that cluster admins can do this based on resource permissions (RBAC).
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console` CRD. 
+We can add 
+`spec.customization.perspectives` which can have list of customization details.
+
+### Workflow Description
+
+To disable perspective for users with limited permissions we need a list of options with a perspective name and the required permissions.
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── name
+|           ├── requiresAccessReview
+|               ├── kind
+|               ├── verb
+└── ...
+```
+
+This might be extendable in the future if we want replace (hide one, show another) a perspective based on a resource permissions:
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── name
+|           ├── requiresAccessReview
+|           |   ├── kind
+|           |   ├── verb
+|           ├── missingAccessReview
+|               ├── kind
+|               ├── verb
+└── ...
+```
+
+Each perspective conforms to the following schema:
+
+// +optional
+- `name: string` , defines the name of perspective to disable. If the name value is not known, it will be ignored.
```

Comment:
As mentioned above, I would recommend to use `id` here instead of name so that it is clear that we compare this with the perspective id from https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/extensions/perspectives.ts#L11-L14

The name/ID should be required:

```suggestion
- `id: string` , defines the id of perspective to disable. Incorrect or unknown ids will be ignored.
```